From: Casey Bodley <cbodley@redhat.com>
Date: Wed, 11 Jan 2017 09:32:59 -0500
Subject: rgw: fix off-by-one in RGWDataChangesLog::get_info

Fixes: http://tracker.ceph.com/issues/18488

Signed-off-by: Casey Bodley <cbodley@redhat.com>
(cherry picked from commit a0974fdcf62e60cf31bc15588e7b718da6f6ade3)

Resolves: rhbz#1330952
---
 src/rgw/rgw_bucket.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rgw/rgw_bucket.cc b/src/rgw/rgw_bucket.cc
index 223e86c..e37351b 100644
--- a/src/rgw/rgw_bucket.cc
+++ b/src/rgw/rgw_bucket.cc
@@ -1626,7 +1626,7 @@ int RGWDataChangesLog::list_entries(const real_time& start_time, const real_time
 
 int RGWDataChangesLog::get_info(int shard_id, RGWDataChangesLogInfo *info)
 {
-  if (shard_id > num_shards)
+  if (shard_id >= num_shards)
     return -EINVAL;
 
   string oid = oids[shard_id];
